packages/perspective-jupyterlab/src/ts/renderer.ts

```diff
@@ -82,21 +82,45 @@ export class PerspectiveDocumentWidget extends DocumentWidget<PerspectiveWidget>
                 // don't handle other mimetypes for now
                 throw "Not handled";
             }
-            perspective
-                .worker()
-                .table(data)
-                .then((table: any) => {
-                    this._table = table;
-                    if (this._psp.viewer.table === undefined) {
-                        // construct new table
-                        this._psp.viewer.load(this._table);
-                    } else {
-                        // replace existing table for whatever reason
-                        this._psp.replace(this._table);
-                    }
-                });
-        } catch {
+            const worker = perspective.worker();
+
+            const table = worker.table(data);
+
+            this._table = table;
+            if (this._psp.viewer.table === undefined) {
+                // construct new table
+                this._psp.viewer.load(this._table);
+            } else {
+                // replace existing table for whatever reason
+                this._psp.replace(this._table);
+            }
+
+            // create a flat view
+            const view = this._table.view();
+            view.on_update(() => {
```

Comment:
Does this work/is actually called in test?  This and other methods are async since #1289 so `this._table` and the return type of `view()` are both `Promise` here, I believe.

Reply:
I'll check again, we have some crazy yarn link stuff to worry about these days